group-coordinator/src/test/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilderTest.java

```diff
@@ -168,20 +170,24 @@ public TargetAssignmentBuilder.TargetAssignmentResult build() {
             });
 
             // Prepare the expected topic metadata.
-            Map<Uuid, AssignmentTopicMetadata> topicMetadata = new HashMap<>();
-            subscriptionMetadata.forEach((topicName, metadata) -> {
-                topicMetadata.put(metadata.id(), new AssignmentTopicMetadata(metadata.numPartitions()));
-            });
+            Map<Uuid, TopicMetadata> topicMetadataMap = new HashMap<>();
+            subscriptionMetadata.forEach((topicName, topicMetadata) ->
+                topicMetadataMap.put(topicMetadata.id(), topicMetadata));
 
             // Prepare the expected assignment spec.
             AssignmentSpec assignmentSpec = new AssignmentSpec(
-                memberSpecs,
-                topicMetadata
+                memberSpecs
+            );
+
+            SubscribedTopicMetadata assignmentTopicMetadata = new SubscribedTopicMetadata(
+                topicMetadataMap
             );
```

Comment:
nit: Could you put this right after the `topicMetadataMap` preparation. They seems to go together.